facade/src/main/java/org/jboss/pnc/facade/providers/BuildFetcher.java

```diff
@@ -0,0 +1,426 @@
+                .skip(toSkip)
+                .limit(pageSize)
+                .collect(Collectors.toList());
+
+        fetchBuildConfigAudited(builds.stream().flatMap(BuildWrapper::buildRecordStream).collect(Collectors.toSet()));
+        return builds.stream().map(BuildWrapper::getBuild).collect(Collectors.toList());
+    }
+
+    /**
+     * Fetch all running builds from build coordinator and cache them.
+     */
+    public void fetchRunningBuilds() {
+        try {
+            for (BuildTask buildTask : buildCoordinator.getSubmittedBuildTasks()) {
+                buildTaskCache.put(buildTask.getId(), buildTask);
+            }
+        } catch (RemoteRequestException | MissingDataException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public boolean buildExists(Base32LongID buildId) {
+        try {
+            Optional<BuildTask> submittedBuildTask = buildCoordinator.getSubmittedBuildTask(buildId.getId());
+            if (submittedBuildTask.isPresent()) {
+                fetchRunningBuilds();
+                return true;
+            }
+            BuildRecord buildRecord = buildRecordRepository.queryById(buildId);
+            if (buildRecord == null) {
+                return false;
+            }
+            buildRecordCache.put(buildRecord.getId(), buildRecord);
+            return true;
+        } catch (RemoteRequestException | MissingDataException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Returns the set of build IDs that are in a Group Build. Empty set means the Group Build does not exist.
+     */
+    public Set<Base32LongID> getGroupBuildContent(long buildConfigSetRecordId) {
+        try {
+            Set<Base32LongID> buildIds = new HashSet<>();
+
+            List<BuildTask> buildTasks = buildCoordinator.getSubmittedBuildTasksBySetId(buildConfigSetRecordId);
+            if (!buildTasks.isEmpty()) {
+                fetchRunningBuilds();
+                buildTasks.stream()
+                        .map(BuildTask::getId)
+                        .map(buildMapper.getIdMapper()::toEntity)
+                        .forEach(buildIds::add);
+            }
+
+            // we need to check DB after build coordinator because of possible race condition
+            BuildConfigSetRecord buildGroup = buildConfigSetRecordRepository.queryById(buildConfigSetRecordId);
+            if (buildGroup == null) {
+                return Set.of();
+            }
+            buildGroup.getBuildRecords().stream().map(BuildRecord::getId).forEach(buildIds::add);
+            return buildIds;
+        } catch (RemoteRequestException | MissingDataException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Fetches the build and all its transitive dependencies to cache.
+     */
+    public void primeBuildDependencies(Base32LongID buildId) {
```

Comment:
what do you think about naming this precacheBuildDependencies or prefetchBuildDependencies? It would be easier to know what the method does. Maybe it would be also good to mention in the name somehow that it also fetches transitive dependencies so that you don't need to go to JavaDoc. Maybe prefetchTransitiveDependencies()?